sormas-backend/src/main/java/de/symeda/sormas/backend/task/TaskFacadeEjb.java

```diff
@@ -763,6 +776,37 @@ private void validate(TaskDto task) throws ValidationRuntimeException {
 		}
 	}
 
+	private String getAssociatedEntityLinkMessagePart(TaskContext taskContext, AbstractDomainObject entity) {
+		if (taskContext.getUrlPattern() == null || entity == null) {
+			return null;
+		}
+
+		String url = getUiUrl(taskContext, entity.getUuid());
+		if (url == null) {
+			return null;
+		}
+
+		String associatedEntityLinkMessage = taskContext.getAssociatedEntityLinkMessage();
+		return String.format(I18nProperties.getString(associatedEntityLinkMessage), url);
+	}
+
+	private String getUiUrl(TaskContext taskContext, String uuid) {
+		if (taskContext.getUrlPattern() == null || uuid == null) {
+			return null;
+		}
+
+		String uiUrl = configFacade.getUiUrl();
+		if (uiUrl == null) {
+			return null;
+		}
+
+		StringBuilder uiUrlBuilder = new StringBuilder(uiUrl);
+		if (!uiUrl.endsWith("/")) {
+			uiUrlBuilder.append("/");
+		}
+		return uiUrlBuilder.append("#!").append(taskContext.getUrlPattern()).append("/data/").append(uuid).toString();
```

Comment:
this (being in the backend module) in combination with the TaskContext (api module) enum extension contain a non-typesafe dependency to the UI, and thus changes in the UI (or the complete change of the UI with the angular UI) can break this
Indeed having a solution for this is out of scope for this ticket but a comment would be good to at least increase the chances of not having an issue in the future